contrib/pyzfs/libzfs_core/_constants.py

```diff
@@ -22,11 +22,15 @@
 
 
 # https://stackoverflow.com/a/1695250
```

Comment:
I realize that this is "just" `contrib` code, but it would be preferable for the comment to explain the code.  We could consider also linking to an external expanded explanation if needed, while weighing the risk of this link becoming stale.  I also realize you aren't adding this link, but if you have time, fixing this would be appreciated.

We could explain why we can't use the built-in python enums (presumably because we want to support pre-2.4 python installs); explain approximately what this code does (creates a new type?); and that this is the canonical way of doing so on old python.  It may then not be necessary to include the external link.

Reply:
I just ~~cargo-culted~~ generalized the existing enum type constructor/generator to do what I needed it to do.
I'd guess the intention behind the link was to point a future developer to practical examples for enum type construction.
FWIW: The stackoverflow page is archived on archive.org ¯\_(ツ)_/¯

I figure future work should replace `_constants.py` with code-generation anyways, so I guess this is not a blocker for merging this feature.

Reply:
This potentially looks like a good candidate to cleanup in another PR if someone is interested.  Python 2 officially [went EOL in January 2020](https://www.python.org/doc/sunset-python-2/) so bumping the minimum supported Python version to 3.4 makes sense.  I tend to agree this doesn't need to block this feature.